workflows/cloudify_system_workflows/idd.py

```diff
@@ -28,6 +29,33 @@ def create(ctx: CloudifyWorkflowContext,
             external_dependencies)
 
 
+def _drop_on_update(idd):
+    """During a dep-update, should this idd be deleted? (and re-created)
+
+    Component and SharedResource IDDs don't need to be deleted, because
+    they will be deleted/created by the Component itself, while uninstalling.
+
+    Custom IDDs, created by the user via the rest-client, ie. not coming from
+    an intrinsic functions, should be kept as well.
+    """
+    creator = idd['dependency_creator']
+    if creator.startswith(
+        ('component.', 'sharedresource.')
+    ):
+        return False
+
+    _, _, last_segment = creator.rpartition('.')
```

Comment:
maybe just
`last_segment = creator.split[-1]` ? can we be sure there are always 3 parts here?

Reply:
the point of partition is that there's always 3 parts yea (possibly the empty string)

I started with split, wrote it wrong (the right way is `creator.rsplit('.', 1)[-1]`) and said "fuck that I'll just always use partition ever, that one is impossible to get wrong" :/